### ISSUE TITLE
jpeg conversion, maybe interesting to you?

### ISSUE THREAD

**Issue by vsaase** [pevers/gdcm-rs#1]
Hi,
just found your repo.
Here is something I did a while ago to decompress jpg DICOM, maybe of interest to you?

https://github.com/vsaase/gdcmjpeg-sys

cheers,
Victor

**Comment by pevers** [comment]
Hi @vsaase ! Thanks a lot!

I'll definitely have a look. I am pretty new to FFI and Rust, so I'll hopefully have some inspiration to fix and finish multi-frame support in https://github.com/Enet4/dicom-rs (even though we want to move away from the GDCM bindings eventually and have a native Rust implementation).

**Comment by vsaase** [comment]
hi, I am new to Rust myself, these were some of the first experiments. 

Somewhere on github I saw a quite short decoder for JPEG-LS, maybe it was charls, but i think to remember something else written purely in C, which should be reasonably easy to port to Rust. 

The build step with gdcm is a bit painful as it requires cmake, ninja, python etc, especially when targetting multiple platforms. long term goal for me would be to have a full dicom library running in WASM.

**Comment by pevers** [comment]
> hi, I am new to Rust myself, these were some of the first experiments.
> 
> Somewhere on github I saw a quite short decoder for JPEG-LS, maybe it was charls, but i think to remember something else written purely in C, which should be reasonably easy to port to Rust.
> 
> The build step with gdcm is a bit painful as it requires cmake, ninja, python etc, especially when targetting multiple platforms. long term goal for me would be to have a full dicom library running in WASM.

Yes that would be awesome! I would love to see this in the browser as well. Especially for large CT scans on top of something like WebGPU.
